tests/pytests/test_parser.py

```diff
@@ -0,0 +1,283 @@
+from includes import *
+from common import *
+
```

Comment:
To make sure we handle deep nesting and right recursion,
I suggest adding such a test to check nesting and also to see we fail gracefully when lemon parser stacksize is exhausted, e.g., with nesting level of 110.

```suggestion

def nest_exp(modifier, term, is_and, i):
    if i == 1:
        return '(@' + modifier + ':' + term + str(i) + ')'
    return '(' + term + str(i) + (' ' if is_and else '|') + nest_exp(modifier, term, is_and, i - 1) + ')'

def testDeepNesting(env):
    nest_level = 100
    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'mod', 'TEXT').ok()
    
    and_exp = nest_exp('mod', 'a', True, nest_level)
    or_exp = nest_exp('mod', 'a', False, nest_level)
    #env.debugPrint(and_exp, force=True)
    #env.debugPrint(or_exp, force=True)
    env.expect('ft.search', 'idx', and_exp).ok()
    env.expect('ft.search', 'idx', or_exp).ok()
    
```

Reply:
Good idea. I also thought about the fact we only check in the intersect and union rules if the left expr has children. Maybe checking for the case that the right expr has children will improve performance